plugins/PromisedScene/Main.js

```diff
@@ -0,0 +1,1220 @@
+
+        // Exit? Break? Return?
+
+        result.name = tpdb_scene_search_data.title;
+      } else {
+        result.name = tpdb_scene_search_data.title;
+      }
+    }
+
+    if (tpdb_scene_search_data.description !== "") {
+      result.description = tpdb_scene_search_data.description;
+    }
+
+    if (tpdb_scene_search_data.date !== "") {
+      result.releaseDate = new Date(tpdb_scene_search_data.date).getTime();
+    }
+
+    if (
+      tpdb_scene_search_data.background.large !== "" &&
+      tpdb_scene_search_data.background.large !== "https://cdn.metadataapi.net/default.png"
+    ) {
+      try {
+        const thumbnailFile = await $createImage(
+          tpdb_scene_search_data.background.large,
+
+          tpdb_scene_search_data.title,
+
+          true
+        );
+
+        result.thumbnail = thumbnailFile.toString();
+      } catch (e) {
+        $log("No thumbnail found");
+      }
+    }
+
+    if (tpdb_scene_search_data.performers !== "") {
+      /* if (args.FemaleOnly) {
+  
+		  let Perf =[]
+  
+		  tpdb_scene_search_data.performers.forEach((Theperson) => {
+  
+			
+  
+			let placeholdername = Freeones(Theperson).then((resultingActor) => {
+  
+			  
+  
+			  if(resultingActor) {
+  
+				return resultingActor
+  
+			  }
+  
+						
+  
+			});
+  
+			
+  
+			if (placeholdername !== undefined) {
+  
+			  Perf.push(placeholdername)
+  
+			}
+  
+					
+  
+		  });
+  
+				
+  
+		  const resolvedFinalArray = await Promise.all(Perf);
+  
+					  
+  
+		  let ResultsActor =[];
+  
+		  resolvedFinalArray.forEach((found) => {
+  
+			if (found !== undefined){
+  
+			  ResultsActor.push(found)
+  
+			}
+  
+		  });
+  
+		  
+  
+		  
+  
+		  
+  
+		} else {
+  
+		  
+  
+		}
+  
+		  */
+
+      result.actors = tpdb_scene_search_data.performers.map((p) => p.name);
+    }
+
+    if (tpdb_scene_search_data.site.name !== "") {
+      result.studio = tpdb_scene_search_data.site.name;
+    }
+
+    $log(" Returning the results");
+
+    return result;
+  }
+
+  // Grabs a list of all the searchable Studios or websites available in TPDB
+
+  async function Grabsites(Metadataapisiteaddress) {
+    try {
+      const ResultTheListofSites = await $axios.get(Metadataapisiteaddress, {
+        validateStatus: false,
+      });
+      const allSites = [];
+
+      if (
+        ResultTheListofSites.status !== 200 ||
+        ResultTheListofSites.data.length === 0 ||
+        testmode.TestSiteunavailable
+      ) {
+        $log(" ERR: TPDB site Not Available OR the API query failed");
+
+        return allSites;
+      }
+
+      const Newtpdb_site_search_content = ResultTheListofSites.data;
+
+      // loops through all of the sites and grabs the "shortname" for the Studio or website
+
+      for (let index = 0; index < Newtpdb_site_search_content.data.length; index++) {
+        const element = Newtpdb_site_search_content.data[index];
+
+        allSites.push(element.short_name);
+      }
+
+      return allSites;
```

Comment:
Minor change for readability

```suggestion
      if (
        ResultTheListofSites.status !== 200 ||
        ResultTheListofSites.data.length === 0 ||
        testmode.TestSiteunavailable
      ) {
        $log(" ERR: TPDB site Not Available OR the API query failed");

        return [];
      }

      const Newtpdb_site_search_content = ResultTheListofSites.data;

      // loops through all of the sites and grabs the "shortname" for the Studio or website

      const allSites = Newtpdb_site_search_content.data.map(el => el.short_name);

      return allSites;
```